database/src/main/resources/db/migration/V1_24__Add_gas_fee_stats.sql

```diff
@@ -0,0 +1,191 @@
+-- Host future and previous migrated data
+CREATE TABLE IF NOT EXISTS tx_single_message_cache
+(
+    id              SERIAL PRIMARY KEY,
+    tx_timestamp    TIMESTAMP             NOT NULL,
+    tx_hash         VARCHAR(64) UNIQUE    NOT NULL,
+    gas_used        INT                   NOT NULL,
+    tx_message_type VARCHAR(128)          NOT NULL,
+    processed       BOOLEAN DEFAULT false NOT NULL
+);
+
+CREATE INDEX IF NOT EXISTS tx_single_message_cache_tx_timestamp_idx
+    ON tx_single_message_cache (tx_timestamp);
+
+CREATE INDEX IF NOT EXISTS tx_single_message_cache_gas_used_idx
+    ON tx_single_message_cache (gas_used);
+
+CREATE INDEX IF NOT EXISTS tx_single_message_cache_tx_message_type_idx
+    ON tx_single_message_cache (tx_message_type);
+
+CREATE INDEX IF NOT EXISTS tx_single_message_cache_processed_idx
+    ON tx_single_message_cache (processed);
+
+-- Aggregated HOURLY data for GAS statistics.
+CREATE TABLE IF NOT EXISTS tx_single_message_gas_stats_day
+(
+    tx_timestamp    TIMESTAMP    NOT NULL,
+    min_gas_used    INT          NOT NULL,
+    max_gas_used    INT          NOT NULL,
+    avg_gas_used    INT          NOT NULL,
+    stddev_gas_used INT          NOT NULL,
+    tx_message_type VARCHAR(128) NOT NULL,
+    PRIMARY KEY (tx_timestamp, tx_message_type)
+);
+
+CREATE TABLE IF NOT EXISTS tx_single_message_gas_stats_hour
+(
+    tx_timestamp    TIMESTAMP    NOT NULL,
+    min_gas_used    INT          NOT NULL,
+    max_gas_used    INT          NOT NULL,
+    avg_gas_used    INT          NOT NULL,
+    stddev_gas_used INT          NOT NULL,
+    tx_message_type VARCHAR(128) NOT NULL,
+    PRIMARY KEY (tx_timestamp, tx_message_type)
+);
+
+-- Migrate historical data to tx_single_message_cache
+WITH single_message_txs AS (
+    SELECT tm.tx_hash,
+           -- avoids adding to GROUP BY
+           string_agg(tc.tx_timestamp::TEXT, ',') AS tx_timestamp,
+           string_agg(tc.gas_used::TEXT, ',')     AS gas_used,
+           string_agg(tmt.type::TEXT, ',')        AS tx_message_type
+    FROM tx_message tm
+             INNER JOIN tx_cache tc
+                        ON tc.id = tm.tx_hash_id
+             INNER JOIN tx_message_type tmt
+                        ON tm.tx_message_type_id = tmt.id
+    GROUP BY tm.tx_hash
+    HAVING count(tm.tx_hash) = 1
+),
+     stats AS (
+         SELECT cast(tx_timestamp AS TIMESTAMP) AS tx_timestamp,
+                tx_hash,
+                cast(gas_used AS INT)           AS gas_used,
+                tx_message_type
+         FROM single_message_txs
+     )
+INSERT
```

Comment:
If you are pre inserting these values when the flyway is run, I would suggest you mark `processed` as true so they dont get re run.

Reply:
We can't do it there because the `insert` for the aggregate tables haven't ran yet. Instead, we can remove the `daily|hourly` insert statements below the `with ...` and do a `call` on the `daily\hourly` procedures after they've been created. Thoughts?

Reply:
can you not add `true AS processed` to the `stats` select?

Reply:
Maybe the `stasts` name is throwing you off but we can not set it to `true` because that `with` clause is moving single tx messages to the `tx_single_message_cache` fact table. The aggregate queries come after that.

Reply:
But the agg queries (in this migration, that initiate the tables) arent using the processed flag at all. If its all being done in this migration, theres no need to set the migration records to false, then true.

Reply:
Ah. Yes. Will update to `true`.